RelationalAI/Client.cs

```diff
@@ -50,19 +50,36 @@ public HttpClient HttpClient
             set { _rest.HttpClient = value; }
         }
 
-        public Task<Database> CreateDatabaseAsync(string database, string engine)
-        {
-            return CreateDatabaseAsync(database, engine, false);
-        }
-
-        public async Task<Database> CreateDatabaseAsync(string database, string engine, bool overwrite)
+        [Obsolete("This method is deprecated, please use CreateDatabaseAsync: engine is no longer required to create a database")]
+        public async Task<Database> CreateDatabaseV1Async(string database, string engine, bool overwrite = false)
         {
             var mode = CreateMode(null, overwrite);
             var transaction = new Transaction(_context.Region, database, engine, mode);
             await _rest.PostAsync(MakeUrl(PathTransaction), transaction.Payload(null), null, transaction.QueryParams());
             return await GetDatabaseAsync(database);
         }
 
+        public async Task<Database> CreateDatabaseAsync(string database, string source = null, string engine = null, bool overwrite = false)
```

Comment:
this is not actually backward compatible, source needs to move to the last argument when being positionally used by client code